operatorcourier/api.py

```diff
@@ -0,0 +1,66 @@
+"""
+operatorcourier.api module
+
+This module implements the api that should be imported when using the operator-courier package
+"""
+
+import os
+import logging
+from tempfile import TemporaryDirectory
+import yaml
+from operatorcourier.build import BuildCmd
+from operatorcourier.validate import ValidateCmd
+from operatorcourier.push import PushCmd
+
+logger = logging.getLogger(__name__)
+
+def build_and_verify(source_dir=None, yamls=None):
+    """Build and verify constructs an operator bundle from a set of files and then verifies it for usefulness and accuracy.
+    It returns the bundle as a string.
+
+    :param source_dir: Path to local directory of yaml files to be read.
+    :param yamls: List of yaml strings to create bundle with
+    """
+
+    yaml_files = []
+
+    if source_dir is not None: 
+        for filename in os.listdir(source_dir):
+            if filename.endswith(".yaml"):
+                with open(source_dir + "/" + filename) as f:
+                    yaml_files.append(f.read())
+    elif yamls is not None:
+        yaml_files = yamls
```

Comment:
why not `yaml_files.extend(yamls)` to respect both options?

If options are mutually exclusive, you should check and raise an exception when both options were used.

Reply:
IMO I am already concerned about the supportability of passing the files as a list (OLM is already implementing a change to use a nested hierarchy with different crds associated with specific versions of csvs), so I would like to keep them separate. I added a check that prevents both parameters from being passed.